python/TestHarness/TestHarness.py

```diff
@@ -464,143 +524,143 @@ def checkPBSEmulator(self):
+
+            # Enter the directory where the test resides
+            saved_cwd = os.getcwd()
+            sys.path.append(os.path.abspath(dirpath))
+            if os.path.exists(dirpath):
+                os.chdir(dirpath)
+            else:
+                print 'Test/s no longer exists!', job_data['test_dir']
+                return
+
+            # Build a Warehouse to hold the MooseObjects
+            warehouse = Warehouse()
+
+            # Build a Parser to parse the objects
+            parser = Parser(self.factory, warehouse)
+
+            # Parse it
+            parser.parse(file)
+
+            # Retrieve the tests from the warehouse
+            testers = warehouse.getAllObjects()
+
+            # launch PBS test
+            self.launchTest(False, file, dirpath, testers, job_data)
+
+            # Repeat...
+            os.chdir(saved_cwd)
+            sys.path.pop()
+
+        # Wait for all qstat processes to finish
+        self.runner.join()
 
-        else:
-            return ('BATCH FILE NOT FOUND', '')
 
     def buildPBSBatch(self, output, tester):
         # Create/Update the batch file
+        pbs_job_id = 0
         if 'command not found' in output:
             tester.setStatus('QSUB NOT FOUND', tester.bucket_fail)
         else:
-            # Get the Job information from the ClusterLauncher
-            results = re.findall(r'JOB_NAME: (\w+) JOB_ID:.* (\d+).*TEST_NAME: (\S+)', output)
-            if len(results) != 0:
-                file_name = self.options.pbs
-                job_list = open(os.path.abspath(os.path.join(tester.specs['executable'], os.pardir)) + '/' + file_name, 'a')
-                for result in results:
-                    (test_dir, job_id, test_name) = result
-                    qstat_command = subprocess.Popen(['qstat', '-f', '-x', str(job_id)], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-                    qstat_stdout = qstat_command.communicate()[0]
-                    # Get the Output_Path from qstat stdout
-                    if qstat_stdout != None:
-                        output_value = re.search(r'Output_Path(.*?)(^ +)', qstat_stdout, re.S | re.M).group(1)
-                        output_value = output_value.split(':')[1].replace('\n', '').replace('\t', '').strip()
-                    else:
-                        job_list.close()
-                        tester.setStatus('QSTAT NOT FOUND', tester.bucket_fail)
-                    # Write job_id, test['test_name'], and Ouput_Path to the batch file
-                    job_list.write(str(job_id) + ':' + test_name + ':' + output_value + ':' + self.options.input_file_name  + '\n')
-                # Return to TestHarness and inform we have launched the job
-                job_list.close()
-                tester.setStatus('LAUNCHED', tester.bucket_pbs)
+            # Get the Job information from the qstat
+            # My regex-fu probably needs help
+            job_id = re.search(r'\n(\d+)\.', output)
```

Comment:
Here is the exact output I have to work with:
```pre
Working Directory: /ssd/milljm/moose/test/tests/kernels/simple_diffusion/job_jason_test\nRunning command: qsub ssd/milljm/moose/test/tests/kernels/simple_diffusion/job_jason_test/jason_test-kernels_simple_diffusion.test.sh\n19818.rod.hpc.inl.gov
```
I originally was trying to regex ^(\d+)\. but that fails because the output gathered has not yet been formatted.